src/astgen.zig

```diff
@@ -443,56 +447,39 @@ pub fn expr(mod: *Module, scope: *Scope, rl: ResultLoc, node: ast.Node.Index) In
         .slice_sentinel => return sliceExpr(mod, scope, rl, tree.sliceSentinel(node)),
 
         .deref => {
-            if (true) @panic("TODO update for zir-memory-layout");
             const lhs = try expr(mod, scope, .none, node_datas[node].lhs);
             const src = token_starts[main_tokens[node]];
-            const result = try addZIRUnOp(mod, scope, src, .deref, lhs);
-            return rvalue(mod, scope, rl, result);
+            const result = try gz.addUnNode(.deref_node, lhs, src);
+            return rvalue(mod, scope, rl, result, src);
         },
         .address_of => {
-            if (true) @panic("TODO update for zir-memory-layout");
+            const src = token_starts[main_tokens[node]];
```

Comment:
As a rule of thumb, all the `token_starts` should be deleted from this file. astgen no longer communicates source locations by absolute byte offsets.